libbeat/publisher/queue/diskqueue/acks_test.go

```diff
@@ -0,0 +1,302 @@
+package diskqueue
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+type addFramesTestStep struct {
+	description      string
+	input            []*readFrame
+	expectedFrameID  frameID
+	expectedPosition *queuePosition
+	expectedSegment  *segmentID
+}
+
+func TestAddFrames(t *testing.T) {
+	// If the done channel is closed, diskQueueACKs.addFrames
+	// should do nothing and immediately return. Otherwise it should:
+	// - add the sizes of all input frames to frameSize
+	// - if any of the input frames are the first frame of their
+	//   respective segment, add their segments to segmentBoundaries
+	// - if the frame with id nextFrameID was among the inputs:
+	//   * advance nextFrameID to the next remaining id that doesn't appear
+	//     in frameSize, and nextPosition to its queuePosition (calculated
+	//     based on the data in frameSize and segmentBoundaries)
+	//   * write the new nextPosition to positionFile
+	//   * remove any frames earlier than nextFrameID from frameSize
+	//   * if we cross any segment boundaries, send the highest such segmentID
+	//     to segmentACKChan (notifying the core loop that they can be
+	//     deleted) and remove earlier segments from segmentBoundaries.
+	dir, err := ioutil.TempDir("", "diskqueue_acks_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+
+	testCases := []struct {
+		name     string
+		frameID  frameID
+		position queuePosition
+		steps    []addFramesTestStep
+	}{
```

Comment:
nit; `cases := map[string]struct{ ... }{ "2-segment test": ... }`

That way the test execution is also randomized, ensuring that we don't have hidden dependencies on state in between tests in general.

Reply:
Done